http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/ResponseHandler.java

```diff
@@ -59,22 +66,24 @@
     protected void channelRead0(ChannelHandlerContext channelContext, HttpObject msg) throws Exception {
         RequestContext requestContext = channelContext.channel().attr(REQUEST_CONTEXT_KEY).get();
 
-
         if (msg instanceof HttpResponse) {
             HttpResponse response = (HttpResponse) msg;
             SdkHttpResponse sdkResponse = SdkHttpFullResponse.builder()
-                                                             .headers(fromNettyHeaders(response.headers()))
-                                                             .statusCode(response.status().code())
-                                                             .statusText(response.status().reasonPhrase())
-                                                             .build();
+                    .headers(fromNettyHeaders(response.headers()))
+                    .statusCode(response.status().code())
+                    .statusText(response.status().reasonPhrase())
+                    .build();
             channelContext.channel().attr(KEEP_ALIVE).set(HttpUtil.isKeepAlive(response));
             requestContext.handler().headersReceived(sdkResponse);
         }
 
         if (msg instanceof StreamedHttpResponse) {
             requestContext.handler().onStream(new PublisherAdapter((StreamedHttpResponse) msg, channelContext, requestContext));
         } else if (msg instanceof FullHttpResponse) {
-            requestContext.handler().onStream(new EmptyRequestPublisher(msg, channelContext));
+            ByteBuf fullContent = ((FullHttpResponse) msg).content();
+            final ByteBuffer bb = copyToByteBuffer(fullContent);
+            fullContent.release();
+            requestContext.handler().onStream(new FullResponseContentPublisher(channelContext, bb));
         }
 
         if (msg instanceof LastHttpContent) {
```

Comment:
If you remove the removePerRequsetHandlers method I think you can combine this with the FullHttpResponse branch.